src/Frontend/src/views/throughputreport/endpoints/DetectedListView.spec.ts

```diff
@@ -0,0 +1,330 @@
+
+      await renderComponent({ source: DataSource.Broker, indicatorOptions: [UserIndicator.NServiceBusEndpoint], columnTitle: columnTitle }, async (driver) => {
+        await driver.setUp(precondition.hasLicensingEndpoints([...[...names].map((name, idx) => ({ name, is_known_endpoint: false, user_indicator: UserIndicator.PlannedToDecommission, max_daily_throughput: idx }))]));
+      });
+
+      const user = userEvent.setup();
+      await user.click(screen.getByRole("button", { name: /Sort by/i }));
+      await user.click(screen.getByRole("link", { name: "name" }));
+
+      const th = screen.getByText(columnTitle);
+      const table = th.parentElement?.parentElement?.parentElement as HTMLTableElement;
+
+      let orderedNames = names.sort((a, b) => a.localeCompare(b));
+      let idx = 0;
+      for (const row of [...table.rows].slice(1)) {
+        expect(row.cells[0].textContent).toBe(orderedNames[idx++]);
+      }
+
+      await user.click(screen.getByRole("button", { name: /Sort by/i }));
+      await user.click(screen.getByRole("link", { name: "name (Descending)" }));
+
+      orderedNames = orderedNames.sort((a, b) => -a.localeCompare(b));
+      idx = 0;
+      for (const row of [...table.rows].slice(1)) {
+        expect(row.cells[0].textContent).toBe(orderedNames[idx++]);
+      }
+    });
+  });
+
+  describe("updating data", () => {
+    const updateLicensingEndpoints =
+      (
+        body: UpdateUserIndicator[] = [
+          <UpdateUserIndicator>{
+            name: "Sender",
+            user_indicator: "",
+          },
+        ]
+      ) =>
+      ({ driver }: SetupFactoryOptions) => {
+        driver.mockEndpoint(`${window.defaultConfig.service_control_url}licensing/endpoints/update`, {
+          body,
+          method: "post",
+          status: 200,
+        });
+        return [];
+      };
+
+    const setup = async () => {
+      const columnTitle = "Special column name";
+
+      const { driver } = await renderComponent({ source: DataSource.Broker, indicatorOptions: [UserIndicator.PlannedToDecommission, UserIndicator.NServiceBusEndpoint], columnTitle: columnTitle }, async (driver) => {
+        await driver.setUp(
+          precondition.hasLicensingEndpoints([
+            { name: `Not set yet`, is_known_endpoint: false, user_indicator: "", max_daily_throughput: 100 },
+            { name: `Set and needs updating`, is_known_endpoint: false, user_indicator: UserIndicator.PlannedToDecommission, max_daily_throughput: 50 },
+          ])
+        );
+        await driver.setUp(updateLicensingEndpoints());
+      });
+
+      const user = userEvent.setup();
+
+      const th = screen.getByText(columnTitle);
+      const table = th.parentElement?.parentElement?.parentElement as HTMLTableElement;
```

Comment:
Consider avoiding navigating the current HTML structure and sticking to aria attributes. The current HTML structure is an implementation detail. Here is an example of where we have accessed tabular data in other tests by using aria attributes:

 This is how it looks while navigating the accessibility tree only (this screenshot was taken using Firefox "inspect accessibility properties" option):

<img width="1018" alt="image" src="https://github.com/user-attachments/assets/87dcde32-d998-44ff-b861-e35e6296629a">

This [link](https://www.w3.org/WAI/ARIA/apg/patterns/table/examples/table/) shows all the different roles available for the "Table" role (table, rowgroup, row, columheader, cell)

The benefit of sticking to aria only is that the implementation has the freedom to change its hierarchy. As long as the aria roles continue to be used, the specific levels of nesting of the hierarchy won't matter, and the test code has a better chance of surviving implementation changes.

Reply:
I fixed this

Reply:
Btw @cquirosj , it does look much nicer the syntax 😄